pennylane/model.py

```diff
@@ -0,0 +1,311 @@
+r"""
+QML Models
+==========
+
+**Module name:** :mod:`pennylane.model`
+
+.. currentmodule:: pennylane.model
+
+This module provides a growing library of functions representing
+common circuit architectures that can be used to easily build more complex
+quantum machine learning models.
+
+For example, you can define and call a circuit-centric quantum classifier
+:cite:`schuld2018circuit` on an arbitrary number of wires and with an
+arbitrary number of blocks in the following way:
+
+.. code-block:: python
+
+    import pennylane as qml
+    from pennylane import numpy as np
+
+    num_wires=4
+    num_blocks=3
+    dev = qml.device('default.qubit', wires=num_wires)
+
+    @qml.qnode(dev)
+    def circuit(weights, x=None):
+        qml.BasisState(x, wires=range(num_wires))
+        qml.model.CircuitCentricClassifier(weights, periodic=True, wires=range(num_wires))
+        return qml.expval.PauliZ(0)
+
+    weights=np.random.randn(num_blocks, num_wires, 3)
+    print(circuit(weights, x=np.array(np.random.randint(0,1,num_wires))))
+
+
+The handy :func:`Interferometer` function can be used to constructs arbitrary interferometers in terms of elementary :class:`~.Beamsplitter` and :class:`~.Rotation`, by means of the scheme from :cite:`clements2016optimal`, specified either via the unitary transformation on the bosonic operators or in terms of lists of beam splitter parameters.
+
+The function :func:`CVNeuralNet` implements the continuous variable neural network architecture from :cite:`killoran2018continuous`. Provided with a suitable array of weights, such models can now be easily constructed and trained with PennyLane.
+
+Summary
+^^^^^^^
+
+.. autosummary::
+  CircuitCentricClassifier
+  CircuitCentricClassifierBlock
+  CVNeuralNet
+  CVNeuralNetLayer
+  Interferometer
+
+Code details
+^^^^^^^^^^^^
+"""
+import logging as log
+import numpy as np
+
+from pennylane.ops import CNOT, Rot, Squeezing, Displacement, Kerr, Rotation, Beamsplitter
+
+log.getLogger()
+
+def CircuitCentricClassifier(weights, periodic=True, ranges=None, imprimitive_gate=CNOT, wires=None):
+    """pennylane.model.CircuitCentricClassifier(weights, periodic=True, ranges=None, imprimitive_gate=CNOT, wires)
+    A circuit-centric classifier circuit.
+
+    Constructs a circuit-centric quantum classifier :cite:`schuld2018circuit`
+    with len(weights) blocks on the given wires with the provided weights.
+    Each element of weights must be a an array of size len(wires)*3.
+
+    Args:
+        weights (array[float]): Number of blocks*len(wires)*3 array of weights
+        periodic (bool): whether to use periodic boundary conditions when
+                         applying imprimitive gates
+        ranges (Sequence[int]): Ranges of the imprimitive gates in the
+                                respective blocks
+        imprimitive_gate (pennylane.ops.Operation): Imprimitive gate to use, defaults to CNOT
+        wires (Sequence[int]): Wires the model should act on
```

Comment:
Just a reminder to double check that the Args block display's correctly in the docs - sphinx can be very finicky when it comes to indentation here.